grafana/grafana
# Provisioning: datasources auto deletion

**What is this feature?**

This PR makes it possible to delete a provisioned datasource by simply deleting it from the provisioning yaml file, or by deleting the yaml file.

**Why do we need this feature?**

Currently, you have to manually add a section to delete data sources, this should all happen automatically. 

**Who is this feature for?**

This feature is for users that provision datasources in yaml files and want to delete them automatically.

**Which issue(s) does this PR fix?**:



Fixes #12875

**Special notes for your reviewer:**

Steps to test it:
- Create a data source provisioning file using api version 1 and add `prune: true` in the file
- Create a provisioned datasource
- Start grafana: The datasource will show up in the list of datasources
- Delete the datasource from the yaml file or delete the provisioning yaml file
- Restart grafana: The datasource should also be deleted

Please check that:
